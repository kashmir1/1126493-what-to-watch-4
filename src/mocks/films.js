export default [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `9.1`,
    rating: `Very good +`,
    votes: `146`,
    description: `Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. `,
    director: `Director: Wes Andreson`,
    starring: [`Starring: Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe and other`],
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    genre: `comedy`,
    year: 2015,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `4.1`,
    rating: `Very poor `,
    votes: `195`,
    description: `Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. `,
    director: `Director: Bill Murray`,
    starring: [`Starring: Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe and other`],
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 3,
    title: `Macbeth`,
    poster: `img/macbeth.jpg`,
    genre: `drama`,
    year: 2014,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `9.3`,
    rating: `Very poor `,
    votes: `119`,
    description: `Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. `,
    director: `Director: Macbeth`,
    starring: [`Starring: Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe and other`],
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 4,
    title: `Aviator`,
    poster: `img/aviator.jpg`,
    genre: `documentary`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `9.2`,
    rating: `Good`,
    votes: `249`,
    description: `Aviator movie description text. Aviator movie description text. Aviator movie description text. Aviator movie description text.`,
    director: `Director: Aviator`,
    starring: [`Starring: Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe and other`],
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 5,
    title: `Moonrise Kingdom`,
    poster: `img/moonrise-kingdom.jpg`,
    genre: `romance`,
    year: 2011,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `9.1`,
    rating: `Good`,
    votes: `232`,
    description: `Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text.`,
    director: `Director: Moonrise Kingdom`,
    starring: [`Starring: Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe and other`],
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 6,
    title: `Midnight Special`,
    poster: `img/midnight-special.jpg`,
    genre: `crime`,
    year: 2016,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `8.4`,
    rating: `Good +`,
    votes: `212`,
    description: `Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text.`,
    director: `Director: Midnight Special`,
    starring: [`Starring: Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe and other`],
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 7,
    title: `War of the Worlds`,
    poster: `img/war-of-the-worlds.jpg`,
    genre: `drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `8.9`,
    rating: `Very good`,
    votes: `201`,
    description: `War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. `,
    director: `Director: War of the Worlds`,
    starring: [`Starring: Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe and other`],
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 8,
    title: `Shutter Island`,
    poster: `img/shutter-island.jpg`,
    genre: `drama`,
    year: 2017,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: `7.9`,
    rating: `Good`,
    votes: `175`,
    description: `Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. `,
    director: `Director: Shutter Island`,
    starring: [`Starring: Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe and other`],
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  }
];


