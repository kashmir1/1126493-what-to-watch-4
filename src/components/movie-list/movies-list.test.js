import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list.jsx';
import films from "../../mocks/testing.js";

const movies = films;

describe(`MoviesList`, () => {
  it(`Render MoviesList`, () => {
    const tree = renderer.create(
        <MoviesList
          movies = {movies}
          onTitleClick = {() => {}}
          onPosterClick = {() => {}}
          moviesList = {films}
        />, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

