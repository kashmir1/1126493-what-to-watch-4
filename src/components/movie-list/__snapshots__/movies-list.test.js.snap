// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList Render MoviesList 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="img/we-need-to-talk-about-kevin.jpg"
        type="video/webm"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page"
        onClick={[Function]}
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="img/what-we-do-in-the-shadows.jpg"
        type="video/webm"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page"
        onClick={[Function]}
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="img/revenant.jpg"
        type="video/webm"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page"
        onClick={[Function]}
      >
        Revenant
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        poster="img/johnny-english.jpg"
        type="video/webm"
        width="280"
      >
        your browser doesn\`t support embedded videos
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page"
        onClick={[Function]}
      >
        Johnny English
      </a>
    </h3>
  </article>
</div>
`;
